user-service/src/main/java/org/orcid/user/web/rest/UserResource.java

```diff
@@ -469,6 +469,48 @@ public boolean getOwner(@PathVariable String salesforceId) {
         LOG.debug("REST request to get Owner for : {}", salesforceId);
         Optional<User> user = userService.getOwnerBySalesforceId(salesforceId);
         return user.isPresent();
-    }     
+    }
+    
+    
+    /**
+     * {@code POST /switch_user} : Switch user 
+     *
+     * @return the {@link ResponseEntity} with status {@code 200 (OK)}.
+     */
+    @PostMapping("/switch_user")
+    public ResponseEntity<Void> switchUser(@RequestParam(value = "username", required = true) String username) {
```

Comment:
@Camelia-Orcid how do you ensure the current user is an admin? it looks like any user could call this endpoint and impersonate other user regardless if it is an admin or not

Reply:
@PreAuthorize("hasRole(\"" + AuthoritiesConstants.ADMIN + "\")") Added